results/cmd/watcher/main.go

```diff
@@ -108,12 +113,58 @@ func (r *reconciler) Reconcile(ctx context.Context, key string) error {
 		r.logger.Errorf("Error converting to proto: %v", err)
 		return err
 	}
-	if _, err := r.client.CreateTaskRunResult(ctx, &pb.CreateTaskRunRequest{
-		TaskRun: p,
-	}); err != nil {
-		r.logger.Error("Error updating TaskRun %s: %v", name, err)
-		return err
-	}
 
+	// Create a TaskRun if it does not exist in results server, update existing one otherwise.
+	annotation := p.GetMetadata().GetAnnotations()
+	if val, ok := annotation["results_id"]; ok {
+		if _, err := r.client.UpdateTaskRunResult(ctx, &pb.UpdateTaskRunRequest{
+			TaskRun:   p,
+			ResultsId: val,
+		}); err != nil {
+			r.logger.Error("Error updating TaskRun %s: %v", name, err)
+			return err
+		}
+		r.logger.Infof("update id ", val)
+	} else {
+		res, err := r.client.CreateTaskRunResult(ctx, &pb.CreateTaskRunRequest{
+			TaskRun: p,
+		})
+		if err != nil {
+			r.logger.Error("Error creating TaskRun %s: %v", name, err)
+			return err
+		}
+
+		// Connect to gke cluster and add results_id to annotations field of TaskRun
```

Comment:
Two improvements here:
1. Creating the kubeconfig can be simplified.
2. We should create the client once and pass it in as a field in the reconciler struct

Reply:
Example for 1: 
-  https://github.com/kubernetes/client-go/blob/master/examples/out-of-cluster-client-configuration/main.go
- https://github.com/kubernetes/client-go/blob/master/examples/in-cluster-client-configuration/main.go

Reply:
done